core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -830,15 +830,16 @@ object IO {
   final def require[E, A](error: E): IO[E, Option[A]] => IO[E, A] =
     (io: IO[E, Option[A]]) => io.flatMap(_.fold[IO[E, A]](IO.fail[E](error))(IO.now[A]))
 
-  final def forkAll[E, A, M[X] <: TraversableOnce[X]](
-    as: M[IO[E, A]]
-  )(implicit cbf: CanBuildFrom[M[IO[E, A]], A, M[A]]): IO[Nothing, Fiber[E, M[A]]] =
-    as.foldLeft(IO.sync[Fiber[E, mutable.Builder[A, M[A]]]](Fiber.point(cbf(as)))) { (asFiberIO, aIO) =>
-        asFiberIO.par(aIO.fork).map {
-          case (asFiber, aFiber) => asFiber.zipWith(aFiber)(_ += _)
-        }
+  final def forkAll[E, A](
+    as: List[IO[E, A]]
+  ): IO[Nothing, Fiber[E, List[A]]] =
+    as.foldLeft(IO.sync[Fiber[E, mutable.Builder[A, ListBuffer[A]]]](Fiber.point(ListBuffer.newBuilder[A]))) {
```

Comment:
We were using the mutable approach b/c of the `CanBuildFrom` abstraction. Since we now have to fold on `List` we can simplify the implementation a lot, also thanks to #118 we can annotate less types:

```scala
    as.foldRight(IO.point(Fiber.point[E, List[A]](List()))) { (aIO, asFiberIO) =>
      asFiberIO.par(aIO.fork).map {
        case (asFiber, aFiber) =>
          asFiber.zipWith(aFiber)((as, a) => a :: as)
      }
    }
```

I'm sure you can apply the same simplification to the other similar methods.

Reply:
Yep, you're right.
That was not my first impl (didn't spotted the `foldRight`)..